packages/palette-charts/src/elements/DataVis/utils/useHasEnteredViewPort.ts

```diff
@@ -1,24 +1,26 @@
-import { useEffect, useState } from "react"
+import { useEffect, useMemo, useState } from "react"
 
 /**
  * Returns true when component enters viewport
  * @param ref reference to component
  */
 export const useHasEnteredViewport = (ref: React.RefObject<HTMLElement>) => {
+
   const [hasEntered, setHasEntered] = useState(false)
+
+  let observer: IntersectionObserver = null
+
+  if (typeof window !== "undefined"){
+    observer = useMemo(() => new IntersectionObserver(
```

Comment:
that hurts one of the hooks' principles, a hook never should be after an if statement or inside of it.